weblate/trans/models/change.py

```diff
@@ -43,7 +43,9 @@ def content(self, prefetch=False):
         return base.filter(action__in=Change.ACTIONS_CONTENT)
 
     def for_category(self, category):
-        return self.filter(component_id__in=category.all_component_ids)
+        if hasattr(category, "component"):
+            return self.filter(component_id__in=category.all_component_ids)
+        return self.filter(category=category)
```

Comment:
It should be more something like this:

```suggestion
        return self.filter(Q(component_id__in=category.all_component_ids) | Q(category=category))
```

Reply:
Hi @nijel, the code change you suggested in the `for_category` function has been included.